tests/test/wasm/test_openmp.cpp

```diff
@@ -1,34 +1,32 @@
 #include <catch2/catch.hpp>
 
-#include "conf/FaasmConfig.h"
-#include "faabric/util/environment.h"
 #include "utils.h"
 
+#include <faabric/proto/faabric.pb.h>
+#include <faabric/scheduler/Scheduler.h>
 #include <faabric/snapshot/SnapshotRegistry.h>
+#include <faabric/util/config.h>
+#include <faabric/util/environment.h>
 #include <faabric/util/func.h>
 
+// Longer timeout to allow longer-running functions to finish
+#define OMP_TEST_TIMEOUT_MS 10000
+
 namespace tests {
 
 void doOmpTestLocal(const std::string& function)
 {
     cleanSystem();
 
-    faabric::snapshot::SnapshotRegistry& reg =
-      faabric::snapshot::getSnapshotRegistry();
-    REQUIRE(reg.getSnapshotCount() == 0);
-
     // Make sure we have enough thread pool capacity
-    conf::FaasmConfig& conf = conf::getFaasmConfig();
-    int32_t defaultPoolSize = conf.moduleThreadPoolSize;
-    conf.moduleThreadPoolSize = 15;
+    faabric::util::SystemConfig& conf = faabric::util::getSystemConfig();
```

Comment:
Should we use `auto&` here as well? (Feel free to ignore)

Reply:
We should use `auto` to reduce verbosity on very commonly used things, but not when it would impact the readability/ understanding of the code. The logger is a good example of where it's used in almost every file, its type name is long, and it's used in almost every file. Adding `auto` on the Scheduler is definitely ok in places where it's used a lot in the same file (e.g. tests), plus we only have one scheduler so you know what type it is. Given that we have two configs (faasm and faabric) I would say it's not ok to do it on `conf` as it might harm the readability, but it's probably not that big a deal either way.